pkg/validate/container_service.go

```diff
@@ -38,7 +38,13 @@ func isValidHostname(h string) bool {
 func ContainerService(new, old *api.ContainerService) (errs []error) {
 	// TODO update validation
 	// TODO are these error messages confusing since they may not correspond with the external model?
-	return validateContainerService(new)
+	if errs := validateContainerService(new); len(errs) > 0 {
+		return errs
+	}
+	if old != nil {
+		return validateUpdateContainerService(new, old)
```

Comment:
@pweil- regarding `MergeConfig` vs `validateUpdateContainerService` vs `ValidateInternal`: 
- I am assuming `validateUpdateContainerService`: is for updating the data model except config
- And  `MergeConfig` is for merging config only?

At what point would the RP need to call `MergeConfig`?

Reply:
> At what point would the RP need to call MergeConfig

'MergeConfig' would be called during updates (so when you have an old and new config), after convert to internal and before `ValidateInternal`.  Step 5 in the doc for update flow.  Since we were unsure who was going to own this impl we figured we'd take a stab at it.  `MergeConfig` is for two items:

1.  ensuring we don't regen certs
2.  dealing with partial manifests received during update

> I am assuming validateUpdateContainerService: is for updating the data model except config
> And MergeConfig is for merging config only?

The RP will always just call `ValidateInternal`.  Since `ValidateInternal` has the old and new config it will call `validateUpdateContainerService` as appropriate (when old != nil).  `validateUpdateContainerService` is simply some update specific validations, no data side effects.

Reply:
@pweil- . Yes,sgtm. I reread the whole `createOrUpdate` in `createorupdate.go` again.